app/models/manageiq/providers/kubevirt/infra_manager/vm/operations/power.rb

```diff
@@ -14,4 +14,12 @@ def raw_stop
       vm.stop
     end
   end
+
+  def raw_reboot_guest
+    ext_management_system.with_provider_connection(:namespace => location) do |connection|
+      # Retrieve the details of the virtual machine:
+      vm = connection.vm(name)
+      vm.restart
```

Comment:
I need to see if `vm.restart` do a "soft" reboot via the guest or a restart via the virtualized hardware.  That would dictate if we call this reboot_guest or reset

Reply:
Based on the API docs it looks like [`VirtualMachine#restart`](https://kubevirt.io/api-reference/v1.4.0/operations.html#_v1restart) is a hardware reset and [`VirtualMachineInstance#softreboot`](https://kubevirt.io/api-reference/v1.4.0/operations.html#_v1softreboot) is what we would call a reboot_guest

Reply:
I added this in power.rb:
![image](https://github.com/user-attachments/assets/3e344fe9-e41c-45d2-a9e4-620d34d500da)
and i got this error:
![image](https://github.com/user-attachments/assets/59369d3e-1bcc-4ccc-b22b-33334dce67af)

Reply:
I think you may need to override these in a subclass of https://github.com/ManageIQ/manageiq/blob/master/app/models/vm/operations/guest.rb

Reply:
I forgot that `with_provider_connection` is still using fog-kubevirt since @nasark's most recent work here was using kubeclient.  I don't see restart or softreboot in fog-kubevirt so we might have an issue there.

Given that we want to move towards kubeclient in the near future as a replacement for fog-kubevirt I think we'd be better off going in that direction rather than adding restart/soft_reboot to fog-kubevirt.

Reply:
Hi @agrare, As suggested I’ve connected to the VM using the following code:

`kubevirt = ext_management_system.parent_manager.connect(:service => "kubernetes", :path => "/apis/subresources.kubevirt.io", :version => "v1")  `

However, I’m not sure which method should be called on this kubevirt instance to perform the restart and reset operation (like `kubevirt.create_virtual_machine_snapshot`). I assume a similar operation should be implemented here using kubeclient. Can you please advise on the correct method I should use for this purpose?

Reply:
Yeah so it looks like kubeclient doesn't pick up on the methods like reset and softreboot because the kubevirt.io API operations don't match the rest of the k8s apis.
I was able to restart and softreboot vms with:
```
client = osv.parent_manager.connect(:service => "kubernetes", :path => "/apis/subresources.kubevirt.io", :version => "v1")
client.rest_client['namespaces/manageiq/virtualmachines/fedora-jade-owl-56/restart'].put({}.to_json, { 'Content-Type' => 'application/json' }.merge(client.get_headers))
client.rest_client['namespaces/manageiq/virtualmachineinstances/fedora-jade-owl-56/softreboot'].put({}.to_json, { 'Content-Type' => 'application/json' }.merge(client.get_headers))
```